### PR TITLE
[NUMBERS-151] performance refine for ArithmeticUtils.pow

### PR DESCRIPTION
**Pull request by XenoAmess**

### CODE REVIEW

**Comment by aherbert** [comment]
You have added no tests. There are a few lines not covered by the existing tests.

**Comment by XenoAmess** [comment]
@aherbert 
> You have added no tests. There are a few lines not covered by the existing tests.

I will go see the coverage and add tests.

BTW.
Should we really throw no exception when we invoke pow(0, 0) here?
It seems...weird...

**Comment by XenoAmess** [comment]
@aherbert 
tests added.

**Comment by aherbert** [comment]
`0^0` is undefined. Here we support returning 1 as described in this Wikipedia article:
[Zero_to_the_power_of_zero](https://en.wikipedia.org/wiki/Zero_to_the_power_of_zero)

Perhaps add a comment:
```
<p>Special cases:
<ul>
<li>{@code k^0} returns {@code 1} (including {@code k=0})
<li>{@code k^1} returns {@code k} (including {@code k=0})
<li>{@code 0^0} returns {@code 1}
<li>{@code 0^e} returns {@code 0}
<li>{@code 1^e} returns {@code 1}
<li>{@code (-1)^e} returns {@code -1 or 1} if {@code e} is odd or even
</ul>
```

File: commons-numbers-core/src/test/java/org/apache/commons/numbers/core/ArithmeticUtilsTest.java [hunk 1]
```diff
@@ -435,6 +435,25 @@ void testIsPowerOfTwo() {
         }
     }
 
+    @Test
+    void testPowOptimize() {
```
Comment (aherbert) [inline commons-numbers-core/src/test/java/org/apache/commons/numbers/core/ArithmeticUtilsTest.java:439]: This should be moved above the `testIsPowerOfTwo` method. I would rename it to `testPowEdgeCases`.

**Review by aherbert — CHANGES_REQUESTED** [review]
A final fix for the new tests and then can you squash all the commits to a single commit.

File: commons-numbers-core/src/main/java/org/apache/commons/numbers/core/ArithmeticUtils.java [hunk 2]
```diff
@@ -266,6 +266,16 @@ public static long lcm(long a, long b) {
     /**
      * Raise an int to an int power.
      *
+     * <p>Special cases:</p>
```
Comment (aherbert) [inline commons-numbers-core/src/main/java/org/apache/commons/numbers/core/ArithmeticUtils.java:269]: You should put the same comment on the `long` version of the pow method.

Comment (XenoAmess) [reply commons-numbers-core/src/main/java/org/apache/commons/numbers/core/ArithmeticUtils.java:269]: @aherbert Ah, I forgot it :)
will do now.

File: commons-numbers-core/src/test/java/org/apache/commons/numbers/core/ArithmeticUtilsTest.java (hunk 1 above)
Comment (XenoAmess) [reply commons-numbers-core/src/test/java/org/apache/commons/numbers/core/ArithmeticUtilsTest.java]: @aherbert done.

**Pull request merged**